docs/references/react/use-auth.mdx

```diff
@@ -9,51 +9,86 @@ The `useAuth()` hook is a convenient way to access the current auth state. This
 
 ## Usage
 
-<CodeBlockTabs options={["Next.js", "React"]}>
-  ```tsx filename="pages/index.tsx"
-  import { useAuth } from '@clerk/nextjs';
-
-  export default function ExternalDataPage() {
-    const { getToken, isLoaded, isSignedIn } = useAuth();
-
-    if (!isLoaded || !isSignedIn) {
-      // You can handle the loading or signed state separately
-      return null;
-    }
-
-    const fetchDataFromExternalResource = async () => {
-      const token = await getToken();
-      // fetch your data
-      return data;
-    };
-
-    return <div>...</div>;
-  }
-
-  ```
-
-  ```tsx filename="external-data-page.tsx"
-  import { useAuth } from '@clerk/clerk-react';
-
-  export default function ExternalDataPage() {
-    const { getToken, isLoaded, isSignedIn } = useAuth();
-
-    if (!isLoaded || !isSignedIn) {
-      // You can handle the loading or signed state separately
-      return null;
-    }
-
-    const fetchDataFromExternalResource = async () => {
-      const token = await getToken();
-      // fetch your data
-      return data;
-    }
-
-    return <div>...</div>;
-  }
-
-  ```
-</CodeBlockTabs>
+<Tabs items={['Next.js', 'React']}>
+  <Tab>
+    <CodeBlockTabs options={["App Router", "Pages Router"]}>
+      ```tsx filename="/app/pageWithData/clientComponent.tsx"
+      "use client";
+      import { useAuth } from '@clerk/nextjs';
+
+      export default function ExternalDataPage() {
+        const [data, setData] = useState<object>({});
+        const { getToken, isLoaded, isSignedIn } = useAuth();
+
+        // Fetch data from a Next.js API route handler
+        useEffect(() => {
+          if (!isLoaded || !isSignedIn) {
+            // You can handle the loading or signed state separately
+            return;
+          }
+
+          // Fetch data without async/await since this is a client component
+          getToken()
+          .then((token) =>
+          fetch('/api/protectedNextRoute', {
+            headers: {Authentication: `Bearer ${token}`}
+          }))
+          .then(response => response.json())
+                .then(data => {
+                  setData(data.data)
+                });
+        }, []);
+
+        return <div>...</div>;
+      }
```

Comment:
Your example is missing React imports! Also, I would abstract the logic out. And you actually can use async/await inside a client component - the component itself just cannot be async. So I've added back in the async/await to have the fetch await for the response from getToken().

```suggestion
"use client";

import * as React from "react";
import { useAuth } from "@clerk/nextjs";

export default function ExternalDataPage() {
  const [data, setData] = React.useState<object>({});
  const { getToken, isLoaded, isSignedIn } = useAuth();
  
  // In this case, /api/example is a Next.js Route Handler
  const fetchData = async () => {
    fetch("/api/example", {
      headers: { Authorization: `Bearer ${await getToken()}` },
    })
      .then((response) => response.json())
      .then((data) => {
        console.log("data", data);
        setData(data);
      });
    return;
  };

  React.useEffect(() => {
    if (!isLoaded || !isSignedIn) {
      // You can handle the loading or signed state separately
    }
    
    fetchData();
  }, []);

  return <div>{JSON.stringify(data, null, 2)}</div>;
}
```

Reply:
@alexisintech Schooled! Thanks for taking the time to teach me something. Much appreciated. Changes have been committed.